app/src/main/java/bg/zahov/app/data/model/state/AddExerciseData.kt

```diff
@@ -0,0 +1,37 @@
+package bg.zahov.app.data.model.state
+
+import bg.zahov.app.data.model.BodyPart
+import bg.zahov.app.data.model.Category
+import bg.zahov.app.data.model.Filter
+import bg.zahov.app.data.model.FilterItem
+
+/**
+ * Data class representing the UI state of the Add Exercise screen.
+ *
+ * @property name The name of the exercise.
+ * @property bodyPartFilters List of filters for selecting a body part.
+ * @property categoryFilters List of filters for selecting a category.
+ * @property uiAddExerciseEventState The current state of the UI, such as showing or hiding dialogs.
+ * @property userMessageId A resource ID for user-facing messages (e.g., error or success).
+ */
+data class AddExerciseData(
+    val name: String = "",
+    val bodyPartFilters: List<FilterItem> = enumValues<BodyPart>().flatMap { bodyPart ->
+        listOf(FilterItem(filter = Filter.BodyPartFilter(bodyPart)))
+    },
+    val categoryFilters: List<FilterItem> = enumValues<Category>().flatMap { category ->
+        listOf(FilterItem(filter = Filter.CategoryFilter(category)))
+    },
+    val uiAddExerciseEventState: AddExerciseEventState = AddExerciseEventState.HideDialog,
```

Comment:
I think it's a better idea to use different ui state fields rather than substates enums https://developer.android.com/topic/architecture/ui-layer/events. There is nothing wrong with this approach as well but as we agreed at the beginning let's stick to google recommendation and later we can explore Mvi events approach